### DISCUSSION TITLE
Apply Color Code to single Orientations in IPF

### DISCUSSION THREAD

**Discussion by MoritzKind** [mtex-toolbox/mtex#572] [Ask Anything] [answered]
Hi MTex Community, 

I'm pretty new to the toolset MTex and I'm trying to takle some problems in plotting orientation Data in IPF figures. 

Plotting the orientations into an IPF is pretty straightfgorward but now I want to color every single orientation with a color code using the additional variable 'relheight' (from 0-1) 
I'm not sure if it is possible to attach the relative height property directly to the file containing the euler angles or if I have to load another file. Additionally it is not really clear to me how to implement a color code in the property 'Markercolor'. 

Below you can see the code and I also uploaded the euler angles file. 

I would be super happy if somebody could give me a hint or tip to solve the problem. 

Thanks and best regards,

Moritz

```
cs = crystalSymmetry('m-3')
ori = orientation.load('eulerY2O3relheight.csv',cs,'ColumnNames',{'phi1','Phi','phi2','relheight'})

plotIPDF(ori,[vector3d.Z,],'MarkerSize',(10),'MarkerColor','red','MarkerEdgeColor','black',)

```

[eulerY2O3relheight.xlsx](https://github.com/mtex-toolbox/mtex/files/5530362/eulerY2O3relheight.xlsx)

**Comment by ralfHielscher** [comment, marked as answer]
You can do

```matlab
plotIPDF(ori,relheight,[vector3d.Z,],'MarkerSize',(10),'MarkerEdgeColor','black')
```
to plot colored orientations. This requires you to have imported `relheight` separately. How to do this you can find at 
https://mtex-toolbox.github.io/S2FunApproximationInterpolation.html

Ralf.

**Reply by MoritzKind** [reply]
Hi Ralf,

thank you for the super fast answer! It worked out pretty well. 

Best,

Moritz
